quota/quota.go

```diff
@@ -313,6 +314,11 @@ func (s *QuotaGroupSet) AddAllNecessaryGroups(grp *Group) {
 		nextParentGrp = nextParentGrp.parentGroup
 	}
 
+	if s.grps[prevParentGrp] {
```

Comment:
is there any real reason to implement this shortcut? we don't expect the trees to be large so we are not really saving any time here with this shortcut

Reply:
that's true to an extent but is still enlarges a factor that then gets multiplied by all the snap services on the system potentially. But my 2nd reason for doing this is to encode the fact that we don't support mixing adding to QGS and manipulating groups at the same time,  and the reason I decided to document that is not that it let's us to the shortcut, but that once I started thinking about it , it's hard to reason exactly what you can do and can't do, so don't do that is easier to reason about.

Reply:
Okay, I decided to just leave this here, it doesn't really hurt anything to have it I suppose, and we should be well testing the code enough to catch any bugs anyways